daphne/src/testing.rs

```diff
@@ -195,32 +205,99 @@ impl DapAggregator<BearerToken> for MockAggregator {
 impl DapHelper<BearerToken> for MockAggregator {
     async fn mark_aggregated(
         &self,
-        _task_id: &Id,
-        _report_shares: &[ReportShare],
+        task_id: &Id,
+        report_shares: &[ReportShare],
     ) -> Result<HashMap<Nonce, TransitionFailure>, DapError> {
-        // Return empty HashMap (for now).
-        // TODO(nakatsuka-y) Implement correct functionality.
-        let early_fails: HashMap<Nonce, TransitionFailure> = HashMap::new();
-        return Ok(early_fails);
+        let mut early_fails: HashMap<Nonce, TransitionFailure> = HashMap::new();
+
+        for report_share in report_shares.iter() {
+            let bucket_info = BucketInfo::new(
+                self.tasks
+                    .get(task_id)
+                    .ok_or_else(|| DapError::fatal("no task found"))?,
+                task_id,
+                &report_share.nonce,
+            );
+
+            let mut report_store_mutex_guard = self
+                .report_store
+                .lock()
+                .map_err(|e| DapError::Fatal(e.to_string()))?;
+
+            let report_store = report_store_mutex_guard.deref_mut();
+
+            if !report_store.contains_key(&bucket_info) {
+                report_store.insert(bucket_info.clone(), ReportStore::new());
+            }
+
+            match report_store
+                .get_mut(&bucket_info)
+                .ok_or_else(|| DapError::fatal("no report store found"))?
+                .process_put_processed(report_share.nonce.clone())
+            {
+                Ok(()) => (),
+                Err(failure_reason) => {
+                    early_fails.insert(report_share.nonce.clone(), failure_reason);
+                }
+            }
+        }
+
+        Ok(early_fails)
     }
 
     async fn put_helper_state(
         &self,
-        _task_id: &Id,
-        _agg_job_id: &Id,
-        _helper_state: &DapHelperState,
+        task_id: &Id,
+        agg_job_id: &Id,
+        helper_state: &DapHelperState,
     ) -> Result<(), DapError> {
-        // Return empty Ok (for now).
-        // TODO(nakatsuka-y) Implement correct functionality.
+        let helper_state_id = HelperStateId {
+            task_id: task_id.clone(),
+            agg_job_id: agg_job_id.clone(),
+        };
+
+        let mut helper_state_store_mutex_guard = self
+            .helper_state_store
+            .lock()
+            .map_err(|e| DapError::Fatal(e.to_string()))?;
+
+        let helper_state_store = helper_state_store_mutex_guard.deref_mut();
+
```

Comment:
This code is correct, however it won't be correct once we add a VDAF that has more than one round of preparation. (Prio3 has one round.) Can we leave a note here so that we don't forget?

```rust 
// NOTE: This code is only correct for VDAFs with exactly one round of preparation.
// For VDAFs with more rounds, the helper state blob will need to be updated here.
```

Reply:
I've added this note to here and another location below this.